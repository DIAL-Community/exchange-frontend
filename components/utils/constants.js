export const DisplayType = {
  LARGE_CARD: 'large-card',
  SMALL_CARD: 'small-card',
  PINNED_CARD: 'pinned-card',
  FEATURED_CARD: 'featured-card',
  HUB_CARD: 'hub-card',
  GRID_CARD: 'grid-card'
}

export const MainDisplayType = {
  LIST: 'list',
  GRID: 'grid'
}

export const ORIGIN_SLUG_EXPANSIONS = {
  'dial': 'Digital Impact Alliance',
  'digital-government-platform-tracker': 'Digital Government Platform Tracker',
  'digital-health-atlas': 'Digital Health Atlas',
  'digital-square': 'Digital Square',
  'dpga': 'Digital Public Goods Alliance',
  'giz': 'GIZ',
  'indiastack': 'India Stack',
  'manually-entered': 'Manually Entered',
  'unicef-covid': 'UNICEF'
}

export const OpportunityStatus = {
  CLOSED: 'CLOSED',
  OPEN: 'OPEN',
  UPCOMING: 'UPCOMING'
}

export const OpportunityType = {
  BID: 'BID',
  BUILDING_BLOCK: 'BUILDING BLOCK',
  INNOVATION: 'INNOVATION',
  TENDER: 'TENDER',
  OTHER: 'OTHER'
}

export const MaturityStatus = {
  DRAFT: 'DRAFT',
  PUBLISHED: 'PUBLISHED'
}

export const CategoryType = {
  DPI: 'DPI',
  FUNCTIONAL: 'FUNCTIONAL'
}

export const MappingStatus = {
  BETA: 'BETA',
  MATURE: 'MATURE',
  SELF_REPORTED: 'SELF-REPORTED',
  VALIDATED: 'VALIDATED'
}

export const LicenseTypeFilter = {
  ALL: 'all_license',
  COMMERCIAL: 'commercial_only',
  OPEN_SOURCE:'oss_only'
}

export const ObjectType = {
  BUILDING_BLOCK: 'BUILDING_BLOCK',
  DATASET: 'OPEN_DATA',
  PRODUCT: 'PRODUCT',
  PROJECT: 'PROJECT',
  OPPORTUNITY: 'OPPORTUNITY',
  ORGANIZATION: 'ORGANIZATION',
  USE_CASE: 'USE_CASE',
  WORKFLOW: 'WORKFLOW',
  MOVE: 'MOVE',
  PLAY: 'PLAY',
  PLAYBOOK: 'PLAYBOOK',
  RUBRIC_CATEGORY: 'RUBRIC_CATEGORY',
  CATEGORY_INDICATOR: 'CATEGORY_INDICATOR',
  CANDIDATE_DATASET: 'CANDIDATE_OPEN_DATA',
  CANDIDATE_ORGANIZATION: 'CANDIDATE_ORGANIZATION',
  CANDIDATE_PRODUCT: 'CANDIDATE_PRODUCT',
  CANDIDATE_RESOURCE: 'CANDIDATE_RESOURCE',
  CANDIDATE_ROLE: 'CANDIDATE_ROLE',
  TAG: 'TAG',
  SECTOR: 'SECTOR',
  URL: 'URL',
  CITY: 'CITY',
  COUNTRY: 'COUNTRY',
  REGION: 'REGION',
  TASK: 'TASK',
  USER: 'USER',
  CONTACT: 'CONTACT',
  RESOURCE: 'RESOURCE',
  RESOURCE_TOPIC: 'RESOURCE_TOPIC',
  SITE_SETTING: 'SITE_SETTING',
  TENANT_SETTING: 'TENANT_SETTING'

}

export const CategoryIndicatorType = {
  NUMERIC: 'numeric',
  SCALE: 'scale',
  BOOLEAN: 'boolean'
}

export const CandidateActionType = {
  APPROVE: 'APPROVE',
  REJECT: 'REJECT'
}

export const DEFAULT_PAGE_SIZE = 8

export const ProductStageType = {
  PILOT: 'pilot',
  SCALING: 'scaling',
  MATURE: 'mature'
}

export const ProductExtraAttributeNames = [
  'Relevance',
  'Impact',
  'Local Ownership',
  'Funders'
]

export const ProductMaturityAttributeNames = [
  'Deployments',
  'Deployed Countries',
  'Active Users',
  'Transactions per month',
  'Annual Revenue',
  'Funding Raised'
]
