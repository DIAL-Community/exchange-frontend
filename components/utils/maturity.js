export const getCategoryIndicatorTypes = (format) => [
  { label: format('shared.categoryIndicator.type.numeric'), value: 'numeric' },
  { label: format('shared.categoryIndicator.type.scale'), value: 'scale' },
  { label: format('shared.categoryIndicator.type.boolean'), value: 'boolean' }
]

export const getCategoryIndicatorScaleOptions = (format) => [
  { label: format('shared.categoryIndicator.type.scale.low'), value: 'low' },
  { label: format('shared.categoryIndicator.type.scale.medium'), value: 'medium' },
  { label: format('shared.categoryIndicator.type.scale.high'), value: 'high' }
]

export const getCategoryIndicatorBooleanOptions = (format) => [
  { label: format('shared.categoryIndicator.type.boolean.true'), value: 't' },
  { label: format('shared.categoryIndicator.type.boolean.false'), value: 'f' }
]

export const getCategoryIndicatorNumericOptions = () =>
  ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10'].map(value => ({ label: value, value }))
