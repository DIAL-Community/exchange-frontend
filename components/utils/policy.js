import { GRAPH_QUERY_CONTEXT_KEY, GRAPH_QUERY_POLICY_SLUG } from '../../lib/apolloClient'

const operationAllowed = (error, operationName) => {
  if (error?.graphQLErrors) {
    const { graphQLErrors } = error
    if (graphQLErrors.length > 0) {
      graphQLErrors.forEach(e => {
        if (e.extensions.operation === operationName) {
          return true
        }
      })
    }
  }

  return false
}

// Return map of operation and boolean value for each operation
// e.g. { editing: true, deleting: true }
export const fetchOperationPolicies = async (client, query, operations, variables = { slug: GRAPH_QUERY_POLICY_SLUG }) => {
  const response = await client.query({
    query,
    variables,
    errorPolicy: 'all',
    context: {
      headers: {
        [GRAPH_QUERY_CONTEXT_KEY]: operations.join(', ')
      }
    }
  })

  const policies = {}
  operations.forEach(operation => {
    policies[operation] = operationAllowed(response.error, operation)
  })

  return policies
}
