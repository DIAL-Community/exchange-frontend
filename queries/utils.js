const MIN_SEARCH_VALUE_CHARACTERS = 2

export const fetchSelectOptions = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}

export const fetchSelectOptionsWithMature = async (client, input, query, fetchedDataCallback, locale = null) => {
  const variables = {
    search: input,
    mature: true
  }

  if (input && input.trim().length < MIN_SEARCH_VALUE_CHARACTERS) {
    return []
  } else if (locale) {
    variables.locale = locale
  }

  const response = await client.query({ query, variables })

  return response.data ? fetchedDataCallback(response.data) : []
}
