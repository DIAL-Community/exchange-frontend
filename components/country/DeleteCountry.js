import { useIntl } from 'react-intl'
import { useCallback, useContext, useState } from 'react'
import { useMutation } from '@apollo/client'
import { useRouter } from 'next/router'
import { DEFAULT_PAGE_SIZE } from '../utils/constants'
import { ToastContext } from '../../lib/ToastContext'
import { useUser } from '../../lib/hooks'
import { DELETE_COUNTRY } from '../shared/mutation/country'
import { PAGINATED_COUNTRIES_QUERY, COUNTRY_DETAIL_QUERY } from '../shared/query/country'
import DeleteButton from '../shared/form/DeleteButton'
import ConfirmActionDialog from '../shared/form/ConfirmActionDialog'

const DeleteCountry = ({ country }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { showSuccessMessage, showFailureMessage } = useContext(ToastContext)

  const router = useRouter()
  const { locale } = router

  const { user } = useUser()

  const [displayConfirmDialog, setDisplayConfirmDialog] = useState(false)
  const toggleConfirmDialog = () => setDisplayConfirmDialog(!displayConfirmDialog)

  const [deleteCountry, { called, reset }] = useMutation(DELETE_COUNTRY, {
    refetchQueries: [{
      query: COUNTRY_DETAIL_QUERY,
      variables: { slug: country.slug }
    }, {
      query: PAGINATED_COUNTRIES_QUERY,
      variables: { search: '', limit: DEFAULT_PAGE_SIZE, offset: 0 }
    }],
    onCompleted: (data) => {
      const { deleteCountry: response } = data
      if (response?.country && response?.errors?.length === 0) {
        setDisplayConfirmDialog(false)
        showSuccessMessage(
          format('toast.delete.success', { entity: format('ui.country.label') }),
          () => router.push(`/${locale}/countries`)
        )
      } else {
        setDisplayConfirmDialog(false)
        showFailureMessage(format('toast.country.delete.failure', { entity: format('ui.country.label') }))
        reset()
      }
    },
    onError: () => {
      setDisplayConfirmDialog(false)
      showFailureMessage(format('toast.country.delete.failure', { entity: format('ui.country.label') }))
      reset()
    }
  })

  const onConfirmDelete = () => {
    if (user) {
      const { userEmail, userToken } = user
      deleteCountry({
        variables: {
          id: country.id
        },
        context: {
          headers: {
            'Accept-Language': locale,
            Authorization: `${userEmail} ${userToken}`
          }
        }
      })
    }
  }

  return (
    <>
      <DeleteButton type='button' onClick={toggleConfirmDialog} />
      <ConfirmActionDialog
        title={format('app.deletingEntity', { entity: country.name })}
        message={format('delete.confirm.message', { entity: format('ui.country.label') })}
        isOpen={displayConfirmDialog}
        onClose={toggleConfirmDialog}
        onConfirm={onConfirmDelete}
        isConfirming={called} />
    </>
  )
}

export default DeleteCountry
