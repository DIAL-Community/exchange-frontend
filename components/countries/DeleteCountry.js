import { useIntl } from 'react-intl'
import { useCallback, useContext, useState } from 'react'
import { useMutation } from '@apollo/client'
import { useRouter } from 'next/router'
import DeleteButton from '../shared/DeleteButton'
import ConfirmActionDialog from '../shared/ConfirmActionDialog'
import { ToastContext } from '../../lib/ToastContext'
import { DELETE_COUNTRY } from '../../mutations/country'
import { COUNTRY_DETAIL_QUERY } from '../../queries/country'
import { useUser } from '../../lib/hooks'

const DeleteCountry = ({ country }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { showToast } = useContext(ToastContext)

  const router = useRouter()
  const { locale } = router

  const { user } = useUser()

  const [isConfirmDialogOpen, setIsConfirmDialogOpen] = useState(false)

  const toggleConfirmDialog = () => {
    setIsConfirmDialogOpen(!isConfirmDialogOpen)
  }

  const [deleteCountry, { called, reset }] = useMutation(DELETE_COUNTRY, {
    refetchQueries: [{
      query: COUNTRY_DETAIL_QUERY,
      variables: { slug: country.slug }
    }],
    onCompleted: () => {
      showToast(
        format('toast.country.delete.success'),
        'success',
        'top-center',
        null,
        () => router.push(`/${locale}/countries`)
      )
      setIsConfirmDialogOpen(false)
    },
    onError: () => {
      showToast(format('toast.country.delete.failure'), 'error', 'top-center')
      setIsConfirmDialogOpen(false)
      reset()
    }
  })

  const onConfirmDelete = () => {
    if (user) {
      const { userEmail, userToken } = user

      deleteCountry({
        variables: {
          id: country.id
        },
        context: {
          headers: {
            'Accept-Language': locale,
            Authorization: `${userEmail} ${userToken}`
          }
        }
      })
    }
  }

  return (
    <>
      <DeleteButton type='button' onClick={toggleConfirmDialog}/>
      <ConfirmActionDialog
        title={format('app.deleting-entity', { entity: country.name })}
        message={format('country.delete.confirm.message')}
        isOpen={isConfirmDialogOpen}
        onClose={toggleConfirmDialog}
        onConfirm={onConfirmDelete}
        isConfirming={called} />
    </>
  )
}

export default DeleteCountry
