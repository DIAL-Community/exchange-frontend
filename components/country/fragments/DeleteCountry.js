import { useCallback, useContext, useState } from 'react'
import { useRouter } from 'next/router'
import { useIntl } from 'react-intl'
import { useMutation } from '@apollo/client'
import { GRAPH_QUERY_CONTEXT } from '../../../lib/apolloClient'
import { ToastContext } from '../../../lib/ToastContext'
import ConfirmActionDialog from '../../shared/form/ConfirmActionDialog'
import DeleteButton from '../../shared/form/DeleteButton'
import { DELETE_COUNTRY } from '../../shared/mutation/country'
import { COUNTRY_DETAIL_QUERY, PAGINATED_COUNTRIES_QUERY } from '../../shared/query/country'
import { DEFAULT_PAGE_SIZE } from '../../utils/constants'

const DeleteCountry = ({ country }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { showSuccessMessage, showFailureMessage } = useContext(ToastContext)

  const router = useRouter()
  const { locale } = router

  const [displayConfirmDialog, setDisplayConfirmDialog] = useState(false)
  const toggleConfirmDialog = () => setDisplayConfirmDialog(!displayConfirmDialog)

  const [deleteCountry, { called, reset }] = useMutation(DELETE_COUNTRY, {
    refetchQueries: [{
      query: COUNTRY_DETAIL_QUERY,
      variables: { slug: country.slug },
      context: {
        headers: {
          ...GRAPH_QUERY_CONTEXT.VIEWING
        }
      }
    }, {
      query: PAGINATED_COUNTRIES_QUERY,
      variables: { search: '', limit: DEFAULT_PAGE_SIZE, offset: 0 },
      context: {
        headers: {
          ...GRAPH_QUERY_CONTEXT.VIEWING
        }
      }
    }],
    onCompleted: (data) => {
      const { deleteCountry: response } = data
      if (response?.country && response?.errors?.length === 0) {
        setDisplayConfirmDialog(false)
        showSuccessMessage(
          format('toast.delete.success', { entity: format('ui.country.label') }),
          () => router.push(`/${locale}/countries`)
        )
      } else {
        setDisplayConfirmDialog(false)
        showFailureMessage(format('toast.country.delete.failure', { entity: format('ui.country.label') }))
        reset()
      }
    },
    onError: () => {
      setDisplayConfirmDialog(false)
      showFailureMessage(format('toast.country.delete.failure', { entity: format('ui.country.label') }))
      reset()
    }
  })

  const onConfirmDelete = () => {
    deleteCountry({
      variables: {
        id: country.id
      },
      context: {
        headers: {
          'Accept-Language': locale
        }
      }
    })
  }

  return (
    <>
      <DeleteButton type='button' onClick={toggleConfirmDialog} />
      <ConfirmActionDialog
        title={format('app.deletingEntity', { entity: country.name })}
        message={format('delete.confirm.message', { entity: format('ui.country.label') })}
        isOpen={displayConfirmDialog}
        onClose={toggleConfirmDialog}
        onConfirm={onConfirmDelete}
        isConfirming={called} />
    </>
  )
}

export default DeleteCountry
