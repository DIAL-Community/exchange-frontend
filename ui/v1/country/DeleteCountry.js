import { useIntl } from 'react-intl'
import { useCallback, useContext, useState } from 'react'
import { useMutation } from '@apollo/client'
import { useRouter } from 'next/router'
import { DEFAULT_PAGE_SIZE, REBRAND_BASE_PATH } from '../utils/constants'
import { ToastContext } from '../../../lib/ToastContext'
import { useUser } from '../../../lib/hooks'
import { DELETE_COUNTRY } from '../shared/mutation/country'
import { PAGINATED_COUNTRIES_QUERY, COUNTRY_DETAIL_QUERY } from '../shared/query/country'
import DeleteButton from '../shared/form/DeleteButton'
import ConfirmActionDialog from '../shared/form/ConfirmActionDialog'

const DeleteCountry = ({ country }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { showToast } = useContext(ToastContext)

  const router = useRouter()
  const { locale } = router

  const { user } = useUser()

  const [displayConfirmDialog, setDisplayConfirmDialog] = useState(false)
  const toggleConfirmDialog = () => setDisplayConfirmDialog(!displayConfirmDialog)

  const [deleteCountry, { called, reset }] = useMutation(DELETE_COUNTRY, {
    refetchQueries: [{
      query: COUNTRY_DETAIL_QUERY,
      variables: { slug: country.slug }
    }, {
      query: PAGINATED_COUNTRIES_QUERY,
      variables: { search: '', limit: DEFAULT_PAGE_SIZE, offset: 0 }
    }],
    onCompleted: (data) => {
      const { deleteCountry: response } = data
      if (response?.country && response?.errors?.length === 0) {
        setDisplayConfirmDialog(false)
        showToast(
          format('toast.country.delete.success'),
          'success',
          'top-center',
          1000,
          null,
          () => router.push(`${REBRAND_BASE_PATH}/countries`)
        )
      } else {
        setDisplayConfirmDialog(false)
        showToast(format('toast.country.delete.failure'), 'error', 'top-center')
        reset()
      }
    },
    onError: () => {
      setDisplayConfirmDialog(false)
      showToast(format('toast.country.delete.failure'), 'error', 'top-center')
      reset()
    }
  })

  const onConfirmDelete = () => {
    if (user) {
      const { userEmail, userToken } = user
      deleteCountry({
        variables: {
          id: country.id
        },
        context: {
          headers: {
            'Accept-Language': locale,
            Authorization: `${userEmail} ${userToken}`
          }
        }
      })
    }
  }

  return (
    <>
      <DeleteButton type='button' onClick={toggleConfirmDialog} />
      <ConfirmActionDialog
        title={format('app.deletingEntity', { entity: country.name })}
        message={format('country.delete.confirm.message')}
        isOpen={displayConfirmDialog}
        onClose={toggleConfirmDialog}
        onConfirm={onConfirmDelete}
        isConfirming={called} />
    </>
  )
}

export default DeleteCountry
