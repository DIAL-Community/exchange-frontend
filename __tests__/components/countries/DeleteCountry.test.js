import { fireEvent, waitFor } from '@testing-library/react'
import DeleteCountry from '../../../components/countries/DeleteCountry'
import { render, mockObserverImplementation } from '../../test-utils'
import CustomMockedProvider from '../../utils/CustomMockedProvider'
import { mockNextAuthUseSession, mockNextUseRouter, statuses } from '../../utils/nextMockImplementation'
import { country } from './data/DeleteCountry'

mockNextUseRouter()
describe('Unit tests for the CountryDelete component.', () => {
  const DELETE_BUTTON_TEST_ID = 'delete-button'
  const CONFIRM_ACTION_DIALOG_TEST_ID = 'confirm-action-dialog'
  const CANCEL_BUTTON_TEST_ID = 'cancel-button'
  const CONFIRM_BUTTON_TEST_ID = 'confirm-button'

  beforeAll(() => {
    mockNextAuthUseSession(statuses.AUTHENTICATED, { canEdit: true })
    window.IntersectionObserver = mockObserverImplementation()
  })

  test('Should open ConfirmActionDialog after clicks "Delete" button.', () => {
    const { getByTestId } = render(
      <CustomMockedProvider>
        <DeleteCountry country={country}/>
      </CustomMockedProvider>
    )
    fireEvent.click(getByTestId(DELETE_BUTTON_TEST_ID))
    expect(getByTestId(CONFIRM_ACTION_DIALOG_TEST_ID)).toBeVisible()
    expect(getByTestId(CONFIRM_ACTION_DIALOG_TEST_ID)).toMatchSnapshot()
  })

  describe('Should close ConfirmActionDialog', () => {
    test('after clicks "Cancel" button.', () => {
      const { getByTestId, queryByTestId } = render(
        <CustomMockedProvider>
          <DeleteCountry country={country}/>
        </CustomMockedProvider>
      )
      fireEvent.click(getByTestId(DELETE_BUTTON_TEST_ID))
      fireEvent.click(getByTestId(CANCEL_BUTTON_TEST_ID))
      expect(queryByTestId(CONFIRM_ACTION_DIALOG_TEST_ID)).not.toBeInTheDocument()
    })

    test('after clicks "Confirm" button', async () => {
      const { getByTestId, queryByTestId } = render(
        <CustomMockedProvider>
          <DeleteCountry country={country}/>
        </CustomMockedProvider>
      )
      fireEvent.click(getByTestId(DELETE_BUTTON_TEST_ID))
      fireEvent.click(getByTestId(CONFIRM_BUTTON_TEST_ID))

      await waitFor(() => expect(queryByTestId(CONFIRM_ACTION_DIALOG_TEST_ID)).not.toBeInTheDocument())
    })
  })
})
