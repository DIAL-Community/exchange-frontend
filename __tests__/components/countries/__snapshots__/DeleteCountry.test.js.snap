// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the CountryDelete component. Should open ConfirmActionDialog after clicks "Delete" button. 1`] = `
<div
  aria-labelledby="headlessui-dialog-title-:r2:"
  aria-modal="true"
  class="fixed z-100"
  data-headlessui-state="open"
  data-testid="confirm-action-dialog"
  id="headlessui-dialog-:r0:"
  role="dialog"
>
  <div
    aria-hidden="true"
    class="fixed inset-0 bg-dial-gray opacity-80"
  />
  <div
    class="fixed inset-0 flex items-center justify-center p-4"
  >
    <div
      class="inline-block w-4/5 md:w-3/5 lg:w-3/5 xl:w-1/2 p-8 transform bg-white shadow-xl rounded-2xl ease-out duration-300 opacity-0 scale-95"
      data-headlessui-state="open"
      id="headlessui-dialog-panel-:r1:"
    >
      <h2
        class="card-title pb-3 text-xl xl:text-2xl text-dial-blue"
        data-headlessui-state="open"
        id="headlessui-dialog-title-:r2:"
      >
        <span
          data-testid="title"
        >
          Deleting country
        </span>
      </h2>
      <span
        class="flex flex-col pb-3"
        data-testid="message"
      >
        Are you sure you want to delete this Country?
      </span>
      <div
        class="flex justify-center sm:justify-end gap-3 text-xl"
      >
        <button
          class="submit-button"
          data-testid="confirm-button"
          type="button"
        >
          Confirm
        </button>
        <button
          class="cancel-button"
          data-testid="cancel-button"
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </div>
</div>
`;
